src/libfetchers/fetch-settings.hh

```diff
@@ -94,6 +93,22 @@ struct FetchSettings : public Config
           empty, the summary is generated based on the action performed.
         )",
         {}, true, Xp::Flakes};
+
+    Setting<bool> trustGitHub{
+        this, true, "trust-github",
+        R"(
+          If enabled (the default), Nix will consider flakes from
+          GitHub and similar providers to be locked if a Git revision
+          is specified,
+          e.g. `github:NixOS/patchelf/7c2f768bf9601268a4e71c2ebe91e2011918a70f`.
+          This requires Nix to trust that the provider will return the
+          correct contents for the specified Git revision.
+
+          If disabled, GitHub flakes are only considered locked if a
+          `narHash` attribute is specified,
+          e.g. `github:NixOS/patchelf/7c2f768bf9601268a4e71c2ebe91e2011918a70f?narHash=sha256-PPXqKY2hJng4DBVE0I4xshv/vGLUskL7jl53roB8UdU%3D`.
+        )"};
```

Comment:
This should be marked experimental; I don't want to need to have it long term.

Reply:
I don't think it should be experimental. It could be removed (made a no-op) in the future if it's no longer needed.

After all, what's the experimental behaviour? The "true" or "false" case?

Reply:
I wasn't there for the rest of the call Monday, but I consider this thing mostly a monkey patch because the other stuff for the fetchers hasn't been done yet.

Reply:
Why wouldn't it be experimental? It only makes sense as part of `fetchTree` anyways, right?

Reply:
It's already implicitly behind `fetch-tree`. But we have a bunch of settings (`access-tokens`, `allow-dirty`, `warn-dirty`, ...) that are also fetcher-specific but not marked as experimental.